builtin/credential/ldap/backend.go

```diff
@@ -195,12 +195,15 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 	// Policies from each group may overlap
 	policies = strutil.RemoveDuplicates(policies, true)
 
-	entityAliasAttribute, err := ldapClient.GetUserAliasAttributeValue(cfg.ConfigEntry, c, username)
-	if err != nil {
-		return "", nil, logical.ErrorResponse(err.Error()), nil, nil
-	}
-	if entityAliasAttribute == "" {
-		return "", nil, logical.ErrorResponse("missing entity alias attribute value"), nil, nil
+	entityAliasAttribute := username
+	if !usernameAsAlias {
+		entityAliasAttribute, err = ldapClient.GetUserAliasAttributeValue(cfg.ConfigEntry, c, username)
+		if err != nil {
+			return "", nil, logical.ErrorResponse(err.Error()), nil, nil
+		}
+		if entityAliasAttribute == "" {
+			return "", nil, logical.ErrorResponse("missing entity alias attribute value"), nil, nil
+		}
 	}
```

Comment:
Nit: Can we flip the check here and do an early return so that we don't end up wrapping the calls within the conditional?

```suggestion
	if usernameAsAlias {
		return username, policies, ldapResponse, allGroups, nil
	}
	
	entityAliasAttribute, err = ldapClient.GetUserAliasAttributeValue(cfg.ConfigEntry, c, username)
	if err != nil {
		return "", nil, logical.ErrorResponse(err.Error()), nil, nil
	}
	if entityAliasAttribute == "" {
		return "", nil, logical.ErrorResponse("missing entity alias attribute value"), nil, nil
	}
```